crates/services/txpool_v2/src/pool.rs

```diff
@@ -228,6 +228,16 @@ where
         Ok(can_store_transaction)
     }
 
+    fn record_transaction_time_in_txpool(tx: &StorageData) {
```

Comment:
Also, we need to remember to record these metrics only if `config.metrics` is `true`.

Reply:
This is done in [ed18ad1](https://github.com/FuelLabs/fuel-core/pull/2321/commits/ed18ad1ab4992a63eb6991ba71908410f5d1e042), although the workflow could be improved. Maybe having  `MetricsRecorder` struct that wraps `TxPoolMetrics` and exposes the functions to record the metrics could work

Reply:
I don't mind the current flow. You're talking about here?
```rs
        best_txs
            .into_iter()
            .inspect(|storage_data| {
                if metrics {
                    Self::record_transaction_time_in_txpool(storage_data)
                }
            })
```
It's pretty easy to read.